src/Microsoft.AspNetCore.OData/EnableNestedPathsAttribute.cs

```diff
@@ -0,0 +1,120 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq;
+using System.Reflection;
+using Microsoft.AspNet.OData.Adapters;
+using Microsoft.AspNet.OData.Common;
+using Microsoft.AspNet.OData.Extensions;
+using Microsoft.AspNet.OData.Query;
+using Microsoft.AspNet.OData.Routing;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Abstractions;
+using Microsoft.AspNetCore.Mvc.Controllers;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.OData.Edm;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// This class defines an attribute that can be applied to an action to allow handling
+    /// of arbitrarily nested paths. The result of the action should be an IQueryable
+    /// or SingleResult. The sequence of property and key accesses in the path will be
+    /// applied to the result of the action through query transformations.
+    /// </summary>
+    [AttributeUsage(AttributeTargets.Method, AllowMultiple = false)]
+    public sealed class EnableNestedPathsAttribute : ActionFilterAttribute
+    {
+        /// <inherit/>
+        public EnableNestedPathsAttribute(): base()
+        {
+            // ensures this filter is executed before [EnableQuery] which has default order 0
+            // since the OnActionExecuted method is executed after the action,
+            // a filter with a lower order will be executed after a filter with a higher order
+            Order = 1;
+        }
+        
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="actionExecutedContext"></param>
+        public override void OnActionExecuted(ActionExecutedContext actionExecutedContext)
+        {
+            if (actionExecutedContext == null)
+            {
+                throw Error.ArgumentNull("actionExecutedContext");
+            }
+
+            HttpRequest request = actionExecutedContext.HttpContext.Request;
+            if (request == null)
+            {
+                throw Error.Argument("actionExecutedContext", SRResources.ActionExecutedContextMustHaveRequest);
+            }
+
+            ActionDescriptor actionDescriptor = actionExecutedContext.ActionDescriptor;
+            if (actionDescriptor == null)
+            {
+                throw Error.Argument("actionExecutedContext", SRResources.ActionContextMustHaveDescriptor);
+            }
+
+
+
+            //base.OnActionExecuted(context);
+            HttpResponse response = actionExecutedContext.HttpContext.Response;
+
+            ObjectResult responseContent = actionExecutedContext.Result as ObjectResult;
```

Comment:
@habbes Remember that return types of `ActionResult`, `ActionResult<T>`, `Queryable<T>` and even `Task<T>` for asynchronous action methods are very common. Will your logic here work with those?
We seem to handle them a certain type of way in `EnableQueryAttribute`
https://github.com/OData/WebApi/blob/5a5c585c139cddc3c63488f03daeb97a4b1ba868/src/Microsoft.AspNetCore.OData/EnableQueryAttribute.cs#L121

Reply:
I think in all those cases, `responseContent.Value` would still be the actual value in your result, i.e. whether you return `Ok(customers)` or just `customers`, the `responseContent.Value` will still be the `customers` object. But let me confirm whether this always holds, including in the case of `Task<T>` which I haven't tested with.